packages/commonwealth/server/controllers/server_groups_methods/update_group.ts

```diff
@@ -1,46 +1,95 @@
 import { AppError } from '../../../../common-common/src/errors';
+import { validateOwner } from '../../util/validateOwner';
+import validateMetadata from '../../util/requirementsModule/validateMetadata';
+import { GroupAttributes, GroupMetadata } from '../../models/group';
+import { sequelize } from '../../database';
 
 const Errors = {
+  InvalidMetadata: 'Invalid metadata',
   InvalidRequirements: 'Invalid requirements',
+  Unauthorized: 'Unauthorized',
+  GroupNotFound: 'Group not found',
 };
 
 export type UpdateGroupOptions = {
   user: UserInstance;
   chain: ChainInstance;
   address: AddressInstance;
-  metadata: any;
-  requirements: Requirement[];
+  groupId: number;
+  metadata?: GroupMetadata;
+  requirements?: Requirement[];
 };
-// TODO: replace with GroupInstance after migration is complete
-export type UpdateGroupResult = {
-  id: number;
-  chain_id: string;
-  metadata: any;
-  requirements: Requirement[];
-}[];
+
+export type UpdateGroupResult = GroupAttributes;
 
 export async function __updateGroup(
   this: ServerChainsController,
-  { requirements }: UpdateGroupOptions
+  { user, chain, groupId, metadata, requirements }: UpdateGroupOptions
 ): Promise<UpdateGroupResult> {
-  // TODO: require community admin
-  const validationError = validateRequirements(requirements);
-  if (validationError) {
-    throw new AppError(validationError.message);
+  const isAdmin = await validateOwner({
+    models: this.models,
+    user,
+    chainId: chain.id,
+    allowMod: true,
+    allowAdmin: true,
+    allowGodMode: true,
+  });
+  if (!isAdmin) {
+    throw new AppError(Errors.Unauthorized);
+  }
+
+  // allow metadata and requirements to be omitted
+  if (typeof metadata !== 'undefined') {
+    const metadataValidationErr = validateMetadata(metadata);
+    if (metadataValidationErr) {
+      throw new AppError(`${Errors.InvalidMetadata}: ${metadataValidationErr}`);
+    }
   }
-  // TODO: delete all existing memberships for group before update
-  return [
-    {
-      id: 1,
-      chain_id: 'ethereum',
-      metadata: {},
-      requirements: [],
+  if (typeof requirements !== 'undefined') {
+    const requirementsValidationErr = validateRequirements(requirements);
+    if (requirementsValidationErr) {
+      throw new AppError(
+        `${Errors.InvalidRequirements}: ${requirementsValidationErr}`
+      );
+    }
+  }
+
+  const group = await this.models.Group.findOne({
+    where: {
+      id: groupId,
+      chain_id: chain.id,
     },
-  ];
+  });
+  if (!group) {
+    throw new AppError(Errors.GroupNotFound);
+  }
+
+  // update the group
+  const toUpdate: Partial<GroupAttributes> = {};
+  if (typeof metadata !== 'undefined') {
+    toUpdate.metadata = metadata;
+  }
+  if (typeof requirements !== 'undefined') {
+    toUpdate.requirements = requirements;
+  }
+
+  await sequelize.transaction(async (transaction) => {
+    // delete all existing memberships for group
+    await this.models.Membership.destroy({
```

Comment:
We only want to destroy memberships if the requirements object gets updated -- pure metadata updates should not destroy memberships.

Reply:
cc @rbennettcw awaiting this change still